stream_alert/shared/lookup_tables/drivers.py

```diff
@@ -0,0 +1,146 @@
+from abc import abstractmethod, ABCMeta
+from stream_alert.shared.logger import get_logger
+from stream_alert.shared.lookup_tables.errors import LookupTablesConfigurationError
+
+LOGGER = get_logger(__name__)
+
+
+# pylint: disable=invalid-name
+class PersistenceDriver(object):
+
+    TYPE_S3 = 's3'
+    TYPE_DYNAMODB = 'dynamodb'
+    TYPE_NULL = 'null'
+    TYPE_EPHEMERAL = 'ephemeral'
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self, configuration):
+        self._configuration = configuration
+
+    @property
+    @abstractmethod
+    def driver_type(self):
+        """Returns a string that describes the type of driver"""
+
+    @property
+    @abstractmethod
+    def id(self):
+        """Returns a unique id for this driver"""
+
+    @abstractmethod
+    def initialize(self):
+        """
+        Configures and initializes this driver
+
+        Initialization is called exactly once, and is should always called BEFORE any other
+        interaction (get/set/commit) is made with the driver.
+        """
+
+    @abstractmethod
+    def commit(self):
+        """
+        Takes any changes and flushes them to remote storage.
+        """
+
+    @abstractmethod
+    def get(self, key, default=None):
+        """Retrieves a key"""
+
+    @abstractmethod
+    def set(self, key, value):
+        """
+        Modifies the value of a key in the LookupTable.
+
+        For LookupTables with remote persistence, you will need to call commit() in order to
+        permanently persist the changes.
+        """
+
+
+def construct_persistence_driver(table_configuration):
+    """
+    Constructs a raw, uninitialized PersistenceDriver from the given configuration.
+
+    :args
+        table_configuration (dict)
+
+    :return
+        PersistenceDriver
+    """
+    import stream_alert.shared.lookup_tables.driver_dynamodb as driver_dynamodb
```

Comment:
why do these imports happen here? it's fine to do these at top of file even if unused

Reply:
There's a cyclical dependency between `drivers.py` (the abstract class, which also contacts the factory method) and `driver_***.py`.

The better way of doing it would be to put the `construct_persistence_river` in a separate module.